src/test/java/org/jenkinsci/plugins/gitclient/GitAPITest.java

```diff
@@ -107,22 +136,29 @@ public void setUpRepositories() throws Exception {
         listener = new hudson.util.LogTaskListener(logger, Level.ALL);
         listener.getLogger().println(LOGGING_STARTED);
 
-        workspace = new WorkspaceWithRepo(repoRootTemp, gitImplName, listener);
+//        workspace = new WorkspaceWithRepo(repoRoot, gitImplName, listener);
+//        workspace = new WorkspaceWithRepo(temporaryDirectoryAllocator.allocate(), gitImplName, listener);
+        workspace = new WorkspaceWithRepo(repo.getRoot(), gitImplName, listener);
 
         testGitClient = workspace.getGitClient();
         testGitDir = workspace.getGitFileDir();
         cliGitCommand = workspace.getCliGitCommand();
-        testGitClient.init();
-        final String userName = "root";
-        final String emailAddress = "root@mydomain.com";
-        cliGitCommand.run("config", "user.name", userName);
-        cliGitCommand.run("config", "user.email", emailAddress);
-        testGitClient.setAuthor(userName, emailAddress);
-        testGitClient.setCommitter(userName, emailAddress);
+    }
+
+    private Collection<String> getBranchNames(Collection<Branch> branches) {
+        return branches.stream().map(Branch::getName).collect(toList());
+    }
+
+    private void assertBranchesExist(Set<Branch> branches, String... names) throws InterruptedException {
+        Collection<String> branchNames = getBranchNames(branches);
+        for (String name : names) {
+            assertThat(branchNames, hasItem(name));
+        }
     }
 
     @Test
     public void testGetRemoteUrl() throws Exception {
+        initializeWorkspace(workspace);
```

Comment:
This looks common to many tests.  Wouldn't it be better to use an `@Before` annotated method to make that happen automatically?  If it is a subset of the total tests, then I think it would be worth splitting the tests into a separate Java file.

```suggestion
        initializeWorkspace(workspace);  // Move this to a separate method with `@Before` annotation
```

Reply:
There was one test which wasn't actually initializing the workspace repo, so to compensate it I made such a function it corresponds to the init function in WorkingArea class.

Reply:
Yes, separate files are good for both those cases.  Having many, many tests in a single test class has caused me a number of challenges when trying to debug and diagnose.

Reply:
Okay, understood. Thanks again @MarkEWaite